src/ints/int10_modes.cpp

```diff
@@ -324,109 +327,267 @@ std::vector<VideoModeBlock> Hercules_Mode = {
-  {0x15,0x00,0x15},{0x15,0x00,0x3f},{0x15,0x2a,0x15},{0x15,0x2a,0x3f},{0x3f,0x00,0x15},{0x3f,0x00,0x3f},{0x3f,0x2a,0x15},{0x3f,0x2a,0x3f},
-  {0x15,0x15,0x00},{0x15,0x15,0x2a},{0x15,0x3f,0x00},{0x15,0x3f,0x2a},{0x3f,0x15,0x00},{0x3f,0x15,0x2a},{0x3f,0x3f,0x00},{0x3f,0x3f,0x2a},
-  {0x15,0x15,0x15},{0x15,0x15,0x3f},{0x15,0x3f,0x15},{0x15,0x3f,0x3f},{0x3f,0x15,0x15},{0x3f,0x15,0x3f},{0x3f,0x3f,0x15},{0x3f,0x3f,0x3f}
+palette_t palette;
+
+constexpr auto num_cga_colors = 16;
+typedef std::array<RGBEntry, num_cga_colors> cga_colors_t;
+
+constexpr RGBEntry black   = {0x00, 0x00, 0x00};
+constexpr RGBEntry blue    = {0x00, 0x00, 0x2a};
+constexpr RGBEntry green   = {0x00, 0x2a, 0x00};
+constexpr RGBEntry cyan    = {0x00, 0x2a, 0x2a};
+constexpr RGBEntry red     = {0x2a, 0x00, 0x00};
+constexpr RGBEntry magenta = {0x2a, 0x00, 0x2a};
+constexpr RGBEntry brown   = {0x2a, 0x15, 0x00};
+constexpr RGBEntry white   = {0x2a, 0x2a, 0x2a};
+
+constexpr RGBEntry br_black   = {0x15, 0x15, 0x15};
+constexpr RGBEntry br_blue    = {0x15, 0x15, 0x3f};
+constexpr RGBEntry br_green   = {0x15, 0x3f, 0x15};
+constexpr RGBEntry br_cyan    = {0x15, 0x3f, 0x3f};
+constexpr RGBEntry br_red     = {0x3f, 0x15, 0x15};
+constexpr RGBEntry br_magenta = {0x3f, 0x15, 0x3f};
+constexpr RGBEntry br_brown   = {0x3f, 0x3f, 0x15};
+constexpr RGBEntry br_white   = {0x3f, 0x3f, 0x3f};
+
+// The canonical CGA palette as emulated by VGA cards.
+static cga_colors_t cga_colors_default = {
+		black, 	  blue,    green,    cyan,    red,    magenta,    brown,    white,
+		br_black, br_blue, br_green, br_cyan, br_red, br_magenta, br_brown, br_white,
 };
 
-static uint8_t mtext_palette[64][3]=
-{
-  {0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},
-  {0x2a,0x2a,0x2a},{0x2a,0x2a,0x2a},{0x2a,0x2a,0x2a},{0x2a,0x2a,0x2a},{0x2a,0x2a,0x2a},{0x2a,0x2a,0x2a},{0x2a,0x2a,0x2a},{0x2a,0x2a,0x2a},
-  {0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},
-  {0x3f,0x3f,0x3f},{0x3f,0x3f,0x3f},{0x3f,0x3f,0x3f},{0x3f,0x3f,0x3f},{0x3f,0x3f,0x3f},{0x3f,0x3f,0x3f},{0x3f,0x3f,0x3f},{0x3f,0x3f,0x3f},
-  {0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},
-  {0x2a,0x2a,0x2a},{0x2a,0x2a,0x2a},{0x2a,0x2a,0x2a},{0x2a,0x2a,0x2a},{0x2a,0x2a,0x2a},{0x2a,0x2a,0x2a},{0x2a,0x2a,0x2a},{0x2a,0x2a,0x2a},
-  {0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},{0x00,0x00,0x00},
-  {0x3f,0x3f,0x3f},{0x3f,0x3f,0x3f},{0x3f,0x3f,0x3f},{0x3f,0x3f,0x3f},{0x3f,0x3f,0x3f},{0x3f,0x3f,0x3f},{0x3f,0x3f,0x3f},{0x3f,0x3f,0x3f} 
+// Emulation of the actual color output of an IBM 5153 monitor (maximum contrast).
+// https://int10h.org/blog/2022/06/ibm-5153-color-true-cga-palette/
+static cga_colors_t cga_colors_ibm5153 = { RGBEntry
```

Comment:
Oh, excellent! color_t is a `std::array`, so all these can be evaluated compile-time instead of being populated at runtime.

`constexpr static cga_colors_t ...`